src/rpc/custom.rs

```diff
@@ -36,4 +41,36 @@ impl CustomService for MissionRpc {
         })?;
         Ok(Response::new(custom::EvalResponse { json }))
     }
+
+    async fn get_magnetic_declination(
+        &self,
+        request: Request<custom::GetMagneticDeclinationRequest>,
+    ) -> Result<Response<custom::GetMagneticDeclinationResponse>, Status> {
+        let position = request.into_inner();
+        let datetime = self
+            .get_scenario_current_time(Request::new(mission::GetScenarioCurrentTimeRequest {}))
```

Comment:
Since these values are not going to change enough to matter over the course of a mission we might consider using the start time instead of the current time.

And since the start time doesn't change is it possible for us to generally call it once at startup and cache the start time when GRPC is started with the mission and then we don't have to call into the MSE at all for it afterwards?

Would save a round-trip to the MSE.

Reply:
Good idea, done in a0476b2d575ae68ef85c3583c1d98ac656da69b2 👍